core/sqf/src/seatrans/tm/hbasetmlib2/hbasetm.cpp

```diff
@@ -860,7 +862,10 @@ int CHbaseTM::createTable(int64 pv_transid,
                     jba_tbldesc,
                     j_keys);
   if (getExceptionDetails(NULL)) {
-     tm_log_write(DTM_TM_JNI_ERROR, SQ_LOG_ERR, (char *)"CHbaseTM::createTable()", (char *)_tlp_error_msg->c_str(), pv_transid);
+	 errstrlen = ((int)_tlp_error_msg->length() < errstrlen) ? (int)_tlp_error_msg->length() : errstrlen;
+	 strncpy(errstr, _tlp_error_msg->c_str(), errstrlen);
+	 errstr[errstrlen -1] = '\0';
```

Comment:
Is it possible at line 865 that errstrlen could be assigned zero? (For example, could (int)_tlp_error_msg->length() return zero?) If so, then this line will clobber the byte just before the beginning of errstr. A reversed buffer overrun.